mypy/typeanal.py

```diff
@@ -634,6 +633,11 @@ def visit_callable_type(self, t: CallableType) -> None:
     def visit_tuple_type(self, t: TupleType) -> None:
         for item in t.items:
             item.accept(self)
+        # if it's not builtins.tuple, then its bases should have tuple[Any]
+        # TODO: put assert here if it's not too slow
+        if isinstance(t.fallback, Instance) and t.fallback.type.fullname() == 'builtins.tuple':
+            fallback_item = UnionType.make_simplified_union(t.items)
```

Comment:
I think it is not a good idea to use unions at this stage, since not all types are analyzed (we might need a separate pass for this in a separate PR).